sale_stock_mto_as_mts_orderpoint_mrp/models/stock_move.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import api, models
+from odoo.tools import groupby
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def all_moves_can_be_bought(self):
+        for move in self:
+            if not move.product_id.purchase_ok:
+                return False
+        return True
+
+    @api.model
+    def product_is_mto(self, product):
+        mto_route = self.env.ref("stock.route_warehouse0_mto")
+        return mto_route in product.route_ids
+
+    def _whatever(self, warehouse):
+        orderpoints_to_procure_ids = []
+
+        if self.all_moves_can_be_bought():
+            return super()._whatever(warehouse)
+
+        seen_moves = self.browse()
+        for bom, bom_move_list in groupby(self, key=lambda m: m.bom_line_id.bom_id):
+            bom_moves = self.browse([move.id for move in bom_move_list])
+            if not bom or not bom_moves:
+                # In case a move doesn't have a bom, do nothing
+                continue
+
+            bom_product = bom.product_id
+            bom_lines = bom.bom_line_ids
+            move_bom_lines = bom_moves.bom_line_id
+            if bom_lines != move_bom_lines:
+                continue  # FIXME
+
+            if not bom_moves.all_moves_can_be_bought() and self.product_is_mto(
+                bom_product
+            ):
```

Comment:
group all your refutations
```suggestion
            if (
                bom_lines != move_bom_lines
                or bom_moves.all_moves_can_be_bought()
                or not self.product_is_mto(bom_product)
            ):
                continue
```

Reply:
I don't like those 4 lines one-liners…